sale_order_line_delivery_state/models/sale_order_line.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2021 ForgeFlow S.L. (https://www.forgeflow.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import float_compare, float_is_zero
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    delivery_state = fields.Selection(
+        [
+            ("no", "No delivery"),
+            ("unprocessed", "Unprocessed"),
+            ("partially", "Partially processed"),
+            ("done", "Done"),
+        ],
+        string="Delivery state",
+        # Compute method have a different name then the field because
+        # the method _compute_delivery_state already exist to compute
+        # the field delivery_set in odoo delivery module
+        compute="_compute_sale_line_delivery_state",
+        store=True,
+    )
+
+    force_delivery_state = fields.Boolean(
+        string="Force delivery state",
+        help=(
+            "Allow to enforce done state of delivery, for instance if some"
+            " quantities were cancelled"
+        ),
+    )
+
+    def _all_qty_delivered(self):
+        """
+        Returns True if line has qty_delivered >= to ordered quantities
+
+        If `delivery` module is installed, ignores the lines with delivery costs
+
+        :returns: boolean
+        """
+        self.ensure_one()
+        precision = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+        return (
+            float_compare(
+                self.qty_delivered, self.product_uom_qty, precision_digits=precision
+            )
+            >= 0
+        )
+
+    def _partially_delivered(self):
+        """
+        Returns True if line has qty_delivered != to 0 and < ordered
+        quantities.
+
+        If `delivery` module is installed, ignores the lines with delivery costs
```

Comment:
Nitpicking: I'm not sure this comment goes here or rather in the _compute_sale_line_delivery_state function where you actually filter out delivery costs

Reply:
Yes! I modified how the delivery lines were treated at one point and forgot to fix the comments. Thank you for spotting it! :)